athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -110,22 +109,35 @@ protected ElasticsearchRecordHandler(AmazonS3 amazonS3, AWSSecretsManager secret
     protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recordsRequest,
                                       QueryStatusChecker queryStatusChecker)
     {
-        logger.info("readWithConstraint - enter\n\nDomain: {}\n\nIndex: {}\n\nMapping: {}\n\nConstraints: {}",
+        logger.info("readWithConstraint - enter\n\nDomain: {}\n\nIndex: {}\n\nMapping: {}",
                 recordsRequest.getTableName().getSchemaName(), recordsRequest.getTableName().getTableName(),
-                recordsRequest.getSchema(), recordsRequest.getConstraints().getSummary());
+                recordsRequest.getSchema());
 
-        String endpoint = ElasticsearchHelper.getDomainEndpoint(recordsRequest.getTableName().getSchemaName());
+        String domain = recordsRequest.getTableName().getSchemaName();
+        String endpoint = recordsRequest.getSplit().getProperty(domain);
         long numRows = 0;
-        AtomicLong numResultRows = new AtomicLong(0);
 
-        if (!endpoint.isEmpty() && queryStatusChecker.isQueryRunning()) {
+        if (endpoint != null && queryStatusChecker.isQueryRunning()) {
```

Comment:
what if the endpoint is indeed null? can we check that above instead of here and throw?

Reply:
Removed check for endpoint != null. Logic will never reach this point if endpoint is null (it will throw an exception in doGetSplits()).